src/main/scala/org/apache/spark/sql/delta/DeltaOptions.scala

```diff
@@ -121,6 +122,37 @@ trait DeltaReadOptions extends DeltaOptionParser {
       throw new IllegalArgumentException(
         s"Please recheck your syntax for '$EXCLUDE_REGEX_OPTION'", e)
   }
+
+  val startingVersion = options.get(STARTING_VERSION_OPTION).map { str =>
+    Try(str.toLong).toOption.filter(_ >= 0).getOrElse{
+      throw DeltaErrors.illegalDeltaOptionException(
+        STARTING_VERSION_OPTION, str, "must be greater than or equal to zero")
+    }
+  }
+
+  val startingTimestamp = options.get(STARTING_TIMESTAMP_OPTION).map { str =>
+    val pattern = "yyyy-MM-dd HH:mm:ss"
```

Comment:
oh, we don't need to add any specific pattern check here. I was hoping we can leave it to Spark to figure it out